### ISSUE TITLE
Regression: Ambassador 1.7.0 reports "error adding listener: multiple filter chains with the same matching rules are defined"

### ISSUE THREAD

**Issue by kflynn** [datawire/ambassador#2957]
This from an Ambassador user running 1.6.2; 1.7.0 fails. In their environment you'll find:

1. An `ambassador` `Module` (via annotation) which sets the `service_port`:

```
---
apiVersion: getambassador.io/v2
kind: Module
name: ambassador
config:
  service_port: 8443
  diagnostics:
    enabled: "true"
```

2: A `tls` `Module` (via annotation):

```
---
apiVersion: getambassador.io/v2
kind: Module
name: tls
config:
  server:
    enabled: True
    redirect_cleartext_from: 8080
  client:
    enabled: False
```

3. Some `TLSContext`s, `Mapping`s, and `Ingress`es.

@kflynn has a snapshot and can see broken `ambassador dump` output from it.

**Comment by jwm** [comment]
If I remove `redirect_cleartext_from: 8080` from the `tls` `Module`, Ambassador starts up just fine.

**Comment by jwm** [comment]
I wound up moving `redirect_cleartext_from` from. our `tls` `Module` onto our various `TLSContext`s, which works around this.

**Comment by khussey** [comment]
This has been fixed in 1.7.4 by reverting the changes for https://github.com/datawire/ambassador/issues/2888 that shipped in 1.7.0.
